src/app/pages/swap/components/swap-selected-asset.layout.tsx

```diff
@@ -0,0 +1,77 @@
+import { FiInfo } from 'react-icons/fi';
+
+import { Box, Stack, Text, color } from '@stacks/ui';
+import { useField } from 'formik';
+
+import { noop } from '@shared/utils';
+
+import { useShowFieldError } from '@app/common/form-utils';
+import { HasChildren } from '@app/common/has-children';
+import { SpaceBetween } from '@app/components/layout/space-between';
+import { Tooltip } from '@app/components/tooltip';
+
+interface SwapSelectedAssetLayoutProps extends HasChildren {
+  caption?: string;
+  onClickHandler?(): void;
+  title: string;
+  tooltipLabel?: string;
+  value: string;
+}
+export function SwapSelectedAssetLayout({
+  caption,
+  children,
+  onClickHandler,
+  title,
+  tooltipLabel,
+  value,
+}: SwapSelectedAssetLayoutProps) {
+  const [, meta] = useField('swapAmountFrom');
+  const showError = useShowFieldError('swapAmountFrom');
```

Comment:
Using formik within a layout component breaks the separation of concerns of the layout only handling  the styling and positioning.

Why not pass in `meta.error` and the other one as a prop?

Reply:
Not sure why I did it this way tbh, will fix.